lib/ApplicationFeatures/EngineEqualityCheckFeature.cpp

```diff
@@ -0,0 +1,92 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Christoph Uhde
+////////////////////////////////////////////////////////////////////////////////
+
+#include "EngineEqualityCheckFeature.h"
+#include "Logger/Logger.h"
+#include "Cluster/ServerState.h"
+#include "Cluster/ClusterInfo.h"
+#include "Cluster/ClusterComm.h"
+#include "StorageEngine/EngineSelectorFeature.h"
+#include "StorageEngine/StorageEngine.h"
+
+using namespace arangodb;
+
+EngineEqualityCheckFeature::EngineEqualityCheckFeature(
+    application_features::ApplicationServer* server)
+    : ApplicationFeature(server, "EngineEqualityCheck") {
+  setOptional(false);
+  requiresElevatedPrivileges(false);
+  startsAfter("Logger");
+}
+
+bool equalStorageEngines(){
+  std::string engineName = EngineSelectorFeature::engineName();
+  auto allEqual = true;
+  auto ci = ClusterInfo::instance();
+  auto cc = ClusterComm::instance();
+
+  // get db servers
+  auto serverIdVector = ci->getCurrentDBServers();
+
+  // prepare requests
+  std::vector<ClusterCommRequest> requests;
+  auto bodyToSend = std::make_shared<std::string>();
+  std::string const url = "/_api/version";
+  for(auto const& id : serverIdVector){
+    requests.emplace_back(id, rest::RequestType::POST, url, bodyToSend);
+  }
+
+  // send requests
+  std::size_t requestsDone = 0;
+  cc->performRequests(requests, 60.0 /*double timeout*/, requestsDone
+                    ,LogTopic("Storage Engine Equality EngineEqualityCheckFeature")
+                    ,false);
+
+  // FIXME make sure all servers responed or fail
+
+  // check answers
+  for (auto const& request : requests){
+    auto const& result = request.result;
+    if(result.status == CL_COMM_RECEIVED) {
+      httpclient::SimpleHttpResult const& simpleResult = *request.result.result;
+      std::string body = simpleResult.getBody().c_str();
+
+      //FIXME extract engine from body
+      LOG_DEVEL << body;
+      auto dbserverEngine = engineName;
```

Comment:
mmh. how can the engines be different here *at all*?

Reply:
They will never be different.